lib/logstash/outputs/elasticsearch/data_stream_support.rb

```diff
@@ -28,7 +28,6 @@ def self.included(base)
       base.extend(Validator)
     end
 
-    # @note assumes to be running AFTER {after_successful_connection} completed, due ES version checks
```

Comment:
This comment looks to be true only for the [assert_es_version_supports_data_streams](https://github.com/logstash-plugins/logstash-output-elasticsearch/pull/1109/files#diff-56fc556089a712f2ab141b29aa8363f8aaf2f3ee65360c538cc42245a0647ca9R144) method.